packages/@lwc/engine-core/src/framework/template.ts

```diff
@@ -118,7 +119,10 @@ function validateLightDomTemplate(template: Template, vm: VM) {
 // TODO [#4078]: Split the implementation between @lwc/engine-dom and @lwc/engine-server
 function buildSerializeExpressionFn(parts?: VStaticPart[]) {
     if (process.env.IS_BROWSER || isUndefined(parts)) {
-        return () => '';
+        return (partToken: string, classAttrToken: string) =>
+            // This will insert the scoped style token as a static class attribute in the fragment
+            // bypassing the need to call applyStyleScoping when mounting static parts.
+            partToken.charAt(0) === STATIC_PART_TOKEN_ID.CLASS ? classAttrToken : '';
```

Comment:
This is a workaround to add the class token in the browser when the fragment is constructed.

Once we can separate the fragment from the SSR environment, we can just use the `${3}` in the browser when a class expression is detected.

Reply:
To elaborate, in the browser environment, there's no issue with serializing the `class` attribute with the fragment since `patchClassAttribute` will add classes to the class list for us.

In the current SSR implementation, the class attribute needs special handling to detect the expression values and then append the scoped token to it (see `serializeClassAttribute` below).

Reply:
This function can be hoisted out into a top-level `const`. it never changes.

Also we should extract `charAt` from `@lwc/shared` instead of calling it directly.

Reply:
Updated! 

1. `charAt` - f76513ac177685e463bab58e2212b6005880c255
2. Hoisting - 67777b8eefb9b7db88b6ffd4c2e1808d62639577